test/Parse/super.swift

```diff
@@ -98,3 +98,26 @@ class Closures : B {
     g()
   }
 }
+
+class ThrowingInitSuperclass {
+  init() throws { }
+}
```

Comment:
Since these are semantic analysis, not parsing, tests, let’s move them to the Sema folder. A new file is fine; no sign of an existing corner dedicated to this feature in Sema.

Reply:
Done! Would you mind reviewing carefully the file header and it's name? I tried my best, but I'm not sure the command on the first line is the ideal one.